lib/_tls_wrap.js

```diff
@@ -247,7 +247,22 @@ function TLSSocket(socket, options) {
   // starting the flow of the data
   this.readable = true;
   this.writable = true;
-  this.read(0);
+
+  // Read on next tick so the caller has a chance to setup listeners
+  process.nextTick(function(self) {
+    // If we were destroyed already don't bother reading
+    if (!self._handle)
+      return;
+
+    // Socket already has some buffered data - emulate receiving it
+    if (socket && socket._readableState && socket._readableState.length) {
+      var buf;
+      while ((buf = socket.read()) !== null)
+        self._handle.receive(buf);
+    }
+
+    self.read(0);
+  }, this);
```

Comment:
Yup! Updated and moved into a named function above the constructor called `initread`